hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/pipeline/SCMPipelineManager.java

```diff
@@ -159,21 +159,14 @@ protected void initializePipelineState() throws IOException {
     }
   }
 
-  @Override
-  public synchronized Pipeline createPipeline(ReplicationType type,
-      ReplicationFactor factor) throws IOException {
-    lock.writeLock().lock();
-    try {
-      Pipeline pipeline = pipelineFactory.create(type, factor);
-      pipelineStore.put(pipeline.getId().getProtobuf().toByteArray(),
-          pipeline.getProtobufMessage().toByteArray());
-      stateManager.addPipeline(pipeline);
-      nodeManager.addPipeline(pipeline);
-      metrics.incNumPipelineAllocated();
+  private void recordMetricsForPipeline(Pipeline pipeline) {
+    switch (pipeline.getType()) {
+    case STAND_ALONE:
       if (pipeline.isOpen()) {
```

Comment:
is this isOpen check duplicated?

Reply:
Works as a kind of protection. Current way to create STAND_ALONE pipeline is to put the state as OPEN indeed. But it's not guaranteed.

Reply:
In this case， metrics.createPerPipelineMetrics(pipeline)  is executed twice for standalone pipeline.

Reply:
Updated